src/shared/cache/ClinicalDataCache.ts

```diff
@@ -37,9 +34,10 @@ function dataToKey(d:any, query:Query) {
 
 async function fetch(
     queries:Query[],
-    samples:SampleIdentifier[],
-    patients:PatientIdentifier[],
-    studyToMutationMolecularProfile:{[studyId:string]:MolecularProfile}
+    samples:Sample[],
+    patients:Patient[],
+    studyToMutationMolecularProfile:{[studyId:string]:MolecularProfile},
+    studyIdToStudy:{[studyId:string]:CancerStudy}
 ) {
     let resultsPerStudyPerAttribute:Object[][][] =
         await Promise.all(queries.map<any>(q=>{
```

Comment:
lets not change this now but this should for sure be a switch statement

Reply:
done